docs/developers/omnichain/overview.md

```diff
@@ -0,0 +1,64 @@
+---
+sidebar_position: 0
+---
+
+# Overview
+
+Omnichain Smart Contracts are contracts deployed on ZetaChain that can use and
+orchestrate assets on connected chains, as well as on ZetaChain. With omnichain
+smart contracts you are able to have a single place of logic that can maintain
+the state of assets and data across all connected chains.
+
+For a contract to be considered omnichain it must inherit from the `zContract`
+interface and implement the `onCrossChainCall` function:
+
+```solidity
+pragma solidity 0.8.7;
+
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/IZRC20.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/zContract.sol";
+
+contract YourContract is zContract {
+    function onCrossChainCall(
+        address zrc20,
+        uint256 amount,
+        bytes calldata message
+    ) external virtual override {
+        bytes32 recipient = abi.decode(message, (bytes32));
+
+        (, uint256 gasFee) = IZRC20(zrc20).withdrawGasFee();
+
+        IZRC20(zrc20).approve(zrc20, gasFee);
+        IZRC20(zrc20).withdraw(abi.encodePacked(recipient), amount - gasFee);
+    }
+}
+```
+
+The contract above is a very simple example of an omnichain contract. Check out
+the [Withdraw tutorial](/developers/omnichain/tutorials/withdraw) for a more
+in-depth example.
+
+An omnichain contract is deployed on ZetaChain and can be called from any
+connected chain.
+
+To call on omnichain contract the only thing a user has to do is send a
+transaction to a connected chain to ZetaChain's TSS address. The transaction
+`amount` becomes available to the sender on ZetaChain and the `data` byte array
+(containing an the omnichain contract `address` and `message`) is used to call
+the omnichain contract by `address` and pass arguments from the `message`.
```

Comment:
we could  add a sentence like `Check out ZRC-20 to learn how to implement this.` that links to the relevant ZRC-20 doc (and/or others)

Reply:
78ad898